src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Builder/OtlpExporterBuilder.cs

```diff
@@ -121,10 +116,6 @@ internal sealed class OtlpExporterBuilder
 
     private static void BindConfigurationToOptions(IServiceCollection services, string name, IConfiguration configuration)
     {
-        Debug.Assert(services != null, "services was null");
-        Debug.Assert(!string.IsNullOrEmpty(name), "name was null or empty");
-        Debug.Assert(configuration != null, "configuration was null");
```

Comment:
I feel like we should keep all of the `Debug.Assert`s being removed on this PR. Those are basically sanity checking that what we declare should not be `null` does in fact have a value. Really nullable annotations are just hints. They don't enforce anything and devs can still make mistakes/decide to override things with `!`. IMO, it is best practice to always `Debug.Assert` those assumptions.

Reply:
I can revert this. At least for.NET Fx I will need to put `!` for usages. Compiler does not undertand it correctly.
Alternative option is to use Guard, but it can be more expensive in execution.

Reply:
Ya I know it's annoying putting all the `!`s just for .NET Fx \ .NET Standard. Definitely prefer adding the `!`s over calling into `Guard` though.

We could drop something like this into `Common.props`:

```xml
	<PropertyGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
                <!-- Disable nullable warnings on older TFMs -->
		<NoWarn>$(NoWarn);8600;8601;8602;8603;8604;8618</NoWarn>
	</PropertyGroup>
```

The idea is to turn off nullable analysis warnings for older TFMs where the IDE/compiler are half-baked. That would eliminate the need for `!`s just for the older things. I didn't do this before because we had some projects which only targeted say `netstandard2.0` but AFAIK everything now has at least one modern `netX.0` target to cover the analysis.

Reply:
In this PR I have reverted Debug assetions without disabling analysis for .NET Fx/ .NET Standard. I think it will be better to do in separate requests. There is a good tooling to automatically remoing unneeded null forgiving operator `!` globally in project/solution. 

See 76a1c9d4b898e894c38439b5429a54aaff3266a3